website/views.py

```diff
@@ -19,15 +40,41 @@ def home(request):
         else:
             messages.success(request, "Failed to log In")
             return redirect('home')
-    else:
-        return render(request,'home.html',{'records':records})
-
-def logout_view(request):
-    logout(request)
-    messages.success(request,"Logged out")
-    return redirect('home')
-def register(request):
-    if request.method == 'POST':
+
+# Function based Logout
+# def logout_view(request):
+#     logout(request)
+#     messages.success(request,"Logged out")
+#     return redirect('home')
+
+class logout_view(View):
+    def get(self,request):
+        logout(request)
+        messages.success(request,"Logged out")
+        return redirect('home')
+
+# Function based Register View
+# def register(request):
+#     if request.method == 'POST':
+#         form = SignUpForm(request.POST)
+#         if form.is_valid():
+#             form.save()
+#             username = form.cleaned_data['username']
+#             password = form.cleaned_data['password1']
+#             user = authenticate(username=username, password=password)
+#             login(request, user)
+#             messages.success(request, "Successfully Registered")
+#             return redirect('home')
+#         else:
+#             return render(request, 'register.html', {'form': form})
+#     else:
+#         form = SignUpForm()
+#         return render(request, 'register.html', {'form': form})
+#     return render(request, 'register.html', {'form': form})
+
+class register(View):
+    template_name='register.html'
+    def post(self,request):
         form = SignUpForm(request.POST)
         if form.is_valid():
             form.save()
```

Comment:
use `form.validated_data['username']` instead of `form.cleaned_data['username']`
also look for the difference b/w cleaned_data and validated_data after `validate()` function